File: app/src/main/java/com/example/rects/SettingLevels.java
//класс создает настройки текущего уровня

package com.example.rects;

public class SettingLevels {

    private static int currentNumLevel;     //номер уровня

    private static int currentMinColor = 0;
    private static int currentMaxColor = 0;
    private static int currentSpace = 0;

    public static void setCurrentSettingLevel(int c) {
        currentNumLevel = c;                                    //номер текуцего уровня
        currentMinColor = SettingGame.getMinColor();            //минимельный цвет текущего уровня

        if(currentNumLevel == 1){
            currentMaxColor = SettingGame.getMaxColorLight();   //максимальный цвет текущего уровня
            currentSpace = SettingGame.getSpaceVeryLight();     //отступы текущего уровня
        }
        if(currentNumLevel == 2){
            currentMaxColor = SettingGame.getMaxColorLight();
            currentSpace = SettingGame.getSpaceLight();
        }
        if(currentNumLevel == 3){
            currentMaxColor = SettingGame.getMaxColorMedium();
            currentSpace = SettingGame.getSpaceMedium();
        }
        if(currentNumLevel == 4){
            currentMaxColor = SettingGame.getMaxColorMedium();
            currentSpace = SettingGame.getSpaceHard();
        }
        if (currentNumLevel == 5){
            currentMaxColor = SettingGame.getMaxColorMedium();
            currentSpace = SettingGame.getSpaceVeryHard();
        }
        if (currentNumLevel == 6){
            currentMaxColor = SettingGame.getMaxColorHard();
            currentSpace = SettingGame.getSpaceVeryHard();
        }
        if (currentNumLevel == 7){
            currentMaxColor = SettingGame.getMaxColorImpossible();
            currentSpace = SettingGame.getSpaceHard();
        }

        SettingLevels.currentMaxColor = currentMaxColor;
    }

    public static int getCurrentMaxColor() {
        return currentMaxColor;
    }

    public static int getCurrentMinColor() {
        return currentMinColor;
    }

    public static int getCurrentSpace() {
        return currentSpace;
    }

    public static int getCurrentNumLevel() {
        return currentNumLevel;
    }
}

File: app/src/main/java/com/example/rects/game/config/SettingLevels.kt
package com.example.rects.game.config

object SettingLevels {
    private const val MIN_LEVEL = 1
    private const val MAX_LEVEL = 7

    private var currentNumLevel = MIN_LEVEL
    private var currentMinColor = SettingGame.getMinColor()
    private var currentMaxColor = SettingGame.getMaxColorLight()
    private var currentSpace = SettingGame.getSpaceVeryLight()

    @JvmStatic
    fun setCurrentSettingLevel(level: Int) {
        currentNumLevel = level.coerceIn(MIN_LEVEL, MAX_LEVEL)
        currentMinColor = SettingGame.getMinColor()

        when (currentNumLevel) {
            1 -> {
                currentMaxColor = SettingGame.getMaxColorLight()
                currentSpace = SettingGame.getSpaceVeryLight()
            }
            2 -> {
                currentMaxColor = SettingGame.getMaxColorLight()
                currentSpace = SettingGame.getSpaceLight()
            }
            3 -> {
                currentMaxColor = SettingGame.getMaxColorMedium()
                currentSpace = SettingGame.getSpaceMedium()
            }
            4 -> {
                currentMaxColor = SettingGame.getMaxColorMedium()
                currentSpace = SettingGame.getSpaceHard()
            }
            5 -> {
                currentMaxColor = SettingGame.getMaxColorMedium()
                currentSpace = SettingGame.getSpaceVeryHard()
            }
            6 -> {
                currentMaxColor = SettingGame.getMaxColorHard()
                currentSpace = SettingGame.getSpaceVeryHard()
            }
            7 -> {
                currentMaxColor = SettingGame.getMaxColorImpossible()
                currentSpace = SettingGame.getSpaceHard()
            }
        }
    }

    @JvmStatic fun getCurrentMaxColor(): Int = currentMaxColor
    @JvmStatic fun getCurrentMinColor(): Int = currentMinColor
    @JvmStatic fun getCurrentSpace(): Int = currentSpace
    @JvmStatic fun getCurrentNumLevel(): Int = currentNumLevel
    @JvmStatic fun getMaxLevel(): Int = MAX_LEVEL
}
